app/ante/ante.go

```diff
@@ -0,0 +1,167 @@
+package ante
+
+import (
+	"fmt"
+	"runtime/debug"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	ibckeeper "github.com/cosmos/ibc-go/v6/modules/core/keeper"
+	evmosante "github.com/evmos/evmos/v12/app/ante"
+	evmosanteevm "github.com/evmos/evmos/v12/app/ante/evm"
+	anteutils "github.com/evmos/evmos/v12/app/ante/utils"
+	evmostypes "github.com/evmos/evmos/v12/types"
+	evmtypes "github.com/evmos/evmos/v12/x/evm/types"
+	evmosvestingtypes "github.com/evmos/evmos/v12/x/vesting/types"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+type HasPermission = func(ctx sdk.Context, accAddr sdk.AccAddress, perm string) bool
+
+func MustCreateHandler(codec codec.BinaryCodec, txConfig client.TxConfig, maxGasWanted uint64, hasPermission HasPermission, accountKeeper evmtypes.AccountKeeper, stakingKeeper evmosvestingtypes.StakingKeeper, bankKeeper evmtypes.BankKeeper, feeMarketKeeper evmosanteevm.FeeMarketKeeper, evmKeeper evmosanteevm.EVMKeeper, ibcKeeper *ibckeeper.Keeper, distrKeeper anteutils.DistributionKeeper) sdk.AnteHandler {
+	ethOpts := evmosante.HandlerOptions{
+		Cdc:                    codec,
+		AccountKeeper:          accountKeeper,
+		BankKeeper:             bankKeeper,
+		ExtensionOptionChecker: evmostypes.HasDynamicFeeExtensionOption,
```

Comment:
should add
`/ethermint.types.v1.ExtensionOptionsWeb3Tx` to allowed extensions

Reply:
I think it's more consistent with the evmos codebase to not do that

Reply:
the legacy one does not have extension options checker
https://github.com/dymensionxyz/evmos/blob/2dcb56724c6cd1610db205af5ee12d84b608828f/app/ante/handler_options.go#L119-L171

Reply:
gonna move convo to https://github.com/dymensionxyz/rollapp-evm/pull/216#discussion_r1592346429